File: pkmnapi-api/src/requests/pokedex_texts.rs
use rocket_okapi::JsonSchema;
use serde::Deserialize;

use crate::requests::base::BaseRequest;

pub type PokedexTextRequest = BaseRequest<PokedexTextRequestType, PokedexTextRequestAttributes>;

impl PokedexTextRequest {
    pub fn get_text(&self) -> &String {
        &self.data.attributes.text
    }
}

#[derive(Debug, Deserialize, JsonSchema)]
#[allow(non_camel_case_types)]
pub enum PokedexTextRequestType {
    pokedex_texts,
}

#[derive(Debug, Deserialize, JsonSchema)]
pub struct PokedexTextRequestAttributes {
    pub text: String,
}

File: clients/js/src/requests/pokedex_texts.ts
import { BaseRequest } from './base';

export type PokedexTextRequest = BaseRequest<PokedexTextRequestType, PokedexTextRequestAttributes>;

enum PokedexTextRequestType {
    pokedex_texts = 'pokedex_texts',
}

export interface PokedexTextRequestAttributes {
    text: string;
}

export const PokedexTextRequestDefault = (attributes: PokedexTextRequestAttributes): PokedexTextRequest => {
    return {
        data: {
            type: PokedexTextRequestType.pokedex_texts,
            attributes,
        },
    };
};
